### PR TITLE
Added support for docker image in docs/docker-buildfiles

### PR DESCRIPTION
**Pull request by rushic24**
FR: https://github.com/AppFlowy-IO/appflowy/issues/163

### CODE REVIEW

**Comment by MikeWallaceDev** [comment]
Hi @rushic24 , thanks so much for the work that you are doing on this! Lots of people have requested a Docker container. 
I looked at the instructions more and something is either not clear to me, or we maybe can do things better. From what I understand the user is still expected to clone the repo first in order to get the files required to build the Docker container, am I right?
If so, we have to do one of three things:
* Include instructions telling the user that they have to clone the repo.
* Change the current instructions and tell them how to get the files without cloning the repo.
* Make the files accessible another way (maybe in the release section??)(or from the wiki?)

Note that I will be adding the instructions and information to the wiki as well, no matter what.

**Comment by MikeWallaceDev** [comment]
@rushic24 ?

**Comment by rushic24** [comment]
Hi, yes sorry, almost forgot. Will update today :).

**Comment by MikeWallaceDev** [comment]
Awesome!  I'll look at it tonight! :)

**Comment by rushic24** [comment]
@MikeWallaceDev added

File: doc/BUILD_ON_DOCKER.md
Comment (MikeWallaceDev) [inline doc/BUILD_ON_DOCKER.md:24]: I was required to put a "sudo" in front of this... Normal?

File: doc/BUILD_ON_DOCKER.md
Comment (MikeWallaceDev) [inline doc/BUILD_ON_DOCKER.md:30]: I was required to put "sudo" here too

```
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/json?all=1&filters=%7B%22label%22%3A%7B%22com.docker.compose.project%3Ddocker-buildfiles%22%3Atrue%7D%7D&limit=0": dial unix /var/run/docker.sock: connect: permission denied
```

File: doc/BUILD_ON_DOCKER.md
Comment (MikeWallaceDev) [inline doc/BUILD_ON_DOCKER.md:31]: The application started, but only a completely white screen appeared. Here is the only output that I got:

```
➜ sudo docker-compose up
[+] Running 1/0
 ⠿ Container docker-buildfiles-app-1  Created                                                                                                                                            0.0s
Attaching to docker-buildfiles-app-1
docker-buildfiles-app-1  | 
docker-buildfiles-app-1  | ** (app_flowy:1): WARNING **: 04:05:32.084: Failed to start Flutter renderer: No available configurations for the given RGBA pixel format
```

**Review by MikeWallaceDev — CHANGES_REQUESTED** [review]
Thanks for this!

MOST of it went fine. It took a long time (that might be added to the docs) but it all worked.

Unfortunately, the application didn't work correctly on my system.  The application did start, but all that appeared is a completely white screen.

Maybe it's the way I started Docker? I haven't done that in a long time. Maybe a quick note on how to start Docker?

All in all, this is fantastic!  Let's get this white screen to work correctly and we'll commit this

Also, since there are comments.... I don't think the /doc directory is the best best for this.  There's a /appflowy/frontend/scripts/ folder, that might be a better place.

Thanks so much for this!

File: doc/BUILD_ON_DOCKER.md
Comment (rushic24) [reply doc/BUILD_ON_DOCKER.md:24]: fixed in readme, you need to add docker to the user group

File: doc/BUILD_ON_DOCKER.md
Comment (rushic24) [reply doc/BUILD_ON_DOCKER.md:31]: Are you using a cloud server for docker and exporting display from there?
Works here for me on my machine. 
image

But if I ssh to my machine from my laptop and follow the procedure then I get your error

Comment (MikeWallaceDev) [reply doc/BUILD_ON_DOCKER.md:43]: Nope, nothing fancy. 
Local machine (Manjaro, up to date)
Followed your script to the letter (copied pasted the instructions)
I'm glad you can at least recreate it on your laptop, it shows that I'm not a complete fool! :D

Comment (rushic24) [reply doc/BUILD_ON_DOCKER.md:43]: No no, I can't do it on my laptop because I don't have enough storage. So I usually ssh my pc from my laptop. The screenshot is by connecting my pc through remote desktop.

Comment (rushic24) [reply doc/BUILD_ON_DOCKER.md:43]: maybe we can ask random people on discord to try and check % of users to whom it works

Comment (MikeWallaceDev) [reply doc/BUILD_ON_DOCKER.md:43]: Well, they won't see this until it is merged.

So what we'll do is, I'll merge it once we're happy with it, then I'll create an issue for my white screen. If more people +1 my issue, then you can look at it.  Personally, I'm all setup, so I won't be using this feature... So it's not a problem if I'm the only one who sees that white screen...

**Review by MikeWallaceDev — APPROVED** [review]
It's all good!  Great job!

LGTM!!

**Comment by LoganTann** [comment]
Hello, are there plans to upload the docker image in dockerhub ?
Also, it may be a good idea to add a link in the readme that points the BUILD_TO_DOCKER.md file

**Comment by MikeWallaceDev** [comment]
There are no plans as of yet.
I just updated the readme to point to the official documentation. I sent in a PR, waiting for it to be accepted.

**Pull request merged**
